### PR TITLE
Feature/yevhen mike updates payments and login

### PR DESCRIPTION
**Pull request by MichaelKatic**
Fixed issues with login logic. Added updates for new storage of fundingsources in object instead of array. Reverted some payment codes and update make payment to only require amount param.

### CODE REVIEW

File: cypress/support/yll/actions.js [hunk 1]
```diff
@@ -620,15 +624,17 @@ const verifyPaymentAccount = ({email = ""} = {}) => {
...
 
         it('Should rollback funding source added time', () => {
-            let method = 'POST';
-            let url = 'https://devapi.yourlandloans.com/users';
-            let body = {
+            const firstFundingSourceId = fundingSources[Object.keys(fundingSources)[0]].id
+
+            const method = 'POST';
+            const url = 'https://devapi.yourlandloans.com/users';
+            const body = {
```
Comment (MichaelKatic) [inline cypress/support/yll/actions.js:631]: Just updated to const instead of let for consistency.

File: cypress/support/yll/actions.js [hunk 2]
```diff
@@ -753,59 +759,41 @@ const makeManualPayment = ({
     });
 };
 
-const makePayment = ({ lenderAccount, loanName, amount }) => {
+const makePayment = ({email = '', loanName = '', amount}) => {
```
Comment (MichaelKatic) [inline cypress/support/yll/actions.js:762]: reverted this change to use "lenderAccount." Again lenders dont make normal payments. "manual" payments are for lenders only.

File: cypress/support/yll/actions.js [hunk 3]
```diff
@@ -753,59 +759,41 @@ const makeManualPayment = ({
...
+        let account;
+        it('Should find account with password and login', () => {
+            getAccount(email).then((foundAccount) => {
+                account = foundAccount
+                loanName = !!loanName ? laonName : account.loans[0];
+                expect(account).to.have.property('password');
+                expect(account.password).not.to.be.empty;
+                login({account});
+            })
+            cy.waitUntil(() => account, {timeout: Cypress.config("defaultCommandTimeout")});
```
Comment (MichaelKatic) [inline cypress/support/yll/actions.js:773]: Another change I had to revert.

File: cypress/support/yll/actions.js [hunk 4]
```diff
@@ -753,59 +759,41 @@ const makeManualPayment = ({
...
-
-            cy.get('div[role="dialog"]')
-                .contains("button", 1)
-                .first()
-                .click({ force: true });
-            cy.contains("button", "OK").click();
+        it(`Should make Payment on Loan in amount ${amount}`, () => {
+            cy.contains('legend', 'Custom Amount').parent().within(() => {
+                cy.get('input[name="paymentAmountRadio"]').click()
+            })
```
Comment (MichaelKatic) [inline cypress/support/yll/actions.js:783]: This radio button update is really all you needed to do for the UI updates.

File: cypress/support/yll/actions.js [hunk 5]
```diff
@@ -753,59 +759,41 @@ const makeManualPayment = ({
...
+            cy.contains('Due Date').parent().contains('Payment Due Now').click();
+            cy.contains('label', 'Notes (Optional)').parent().get('input[placeholder="Extra January Payment"]').type('Cypress automated test payment.');
 
+            cy.contains('button', 'Review Payment Details').click();
             expectWindowAlert({
-                trigger: () => cy.get("form").submit(), // Have to use direct form submit instead.
-                text: "Manual Payment Created",
+                //trigger: () => cy.contains('button', 'Confirm Payment of $1,234.56').click() //Cannot click thie button? "Element is being covered by another element"
+                trigger: () => cy.get('form').submit(), // Have to use direct form submit instead.
+                text: 'Payment Processed',
```
Comment (MichaelKatic) [inline cypress/support/yll/actions.js:796]: ^ These changes are all just reverted from the development branch.

File: cypress/support/yll/util.js [hunk 6]
```diff
@@ -9,35 +9,57 @@ const login = ({account, forwardUrl = '', loginUrl = '', logoutFirst=true, force
         Cypress.on('uncaught:exception', (err, runnable) => {
             return false; //Dangeriously ignoring all uncaught exceptions
         })
+
```
Comment (MichaelKatic) [inline cypress/support/yll/util.js:12]: Updates to make login wait for logout to finish.

File: cypress/e2e/trigger/borrower/signup-pay-flow.cy.js [hunk 7]
```diff
@@ -45,8 +45,6 @@ describe('Borrower Flow - Sign up Through First Payment', () => {
     verifyPaymentAccount();
 
     makePayment({
-        loanName: loan.name, 
-        lenderAccount: accounts.lender, 
         amount: amount
```
[the remark this answers is not in the document]
Comment (zhen1asemen1uk) [reply cypress/e2e/trigger/borrower/signup-pay-flow.cy.js:48]: Okay, I got it, thank you

File: cypress.config.js [hunk 8]
```diff
@@ -18,7 +18,7 @@ module.exports = defineConfig({
   video: true,
   retries: {
     runMode: 2, // Configure retry attempts for `cypress run`
-    openMode: 1 // Configure retry attempts for `cypress open`
+    openMode: 0 // Configure retry attempts for `cypress open`
```
[the remark this answers is not in the document]
Comment (zhen1asemen1uk) [reply cypress.config.js:21]: I got an error in Gmail request, and a secondary try help me with it

**Pull request merged**